File: dev/RStudio/src/events/fairy.R
# TODO: Refactor

#Function for fairy event
fairy_event <- function(castle_data,player){
  #Add space between previously printed screen and new screen
  castle_data$castle[player$movement_coordinate] <- player$encountered_object
  display_array(castle_data = castle_data,player = player,game_sequence = "non-battle")
  cat("You encountered a fairy!")
  new_line(2)
  #Player mana and hp capped at 100
  if(player$hp == 100 & player$mana == 100){
    cat("Your HP and mana are already full. Come back later.")
    #Add pause to allow player to read information
  }else{
    if(player$hp == 100 & player$mana < 100){
      cat("Your mana was fully restored!.")
      player$mana <- 100
    }else if(player$hp < 100 & player$mana == 100){
      cat("Your HP was fully restored!")
      player$hp <- 100
    }else{
      cat("Your HP and mana were fully restored!")
      player$mana <- 100
      player$hp <- 100
    }
    
    castle_data$dataframe[player$castle_dataframe_row,"hp"] <- 0
  }
  #Add pause to allow player to read information
  Sys.sleep(1.5)
  #Adding back door 
  castle_data$castle[player$movement_coordinate] <- "\U1F6AA"
  if(castle_data$dataframe[player$castle_dataframe_row,"hp"] == 0){
    #Adding zero to dataframe so that fairy event is deactivated
    castle_data$dataframe[player$castle_dataframe_row,"hp"] <- 0
  }
  #Return information
  fairy_event_output <- c(castle_data,player)
  return(fairy_event_output)
}

File: dev/Python/_utils/events/fairy.py
# TODO: Refactor
import time
from display import *
from inventory import *
#Functions for various events
def fairy_event(castle, player, castle_info):
    castle[player.movement_coordinate] = player.encountered_object
    display_array(castle = castle, game_sequence = 'non-battle', player = player)
    print('You encountered a fairy!')
    new_line(1)
    if all([player.hp == 100, player.mana == 100]):
        print('Your HP and mana are already full. Come back later.')
        time.sleep(1.5)
    else:
        if player.hp == 100 & player.mana < 100:
            print('Your mana was fully restored!')
            player.mana = 100
        elif player.hp < 100 & player.mana == 100:
            print('Your HP was fully restored!')
            player.hp = 100
        else:
            player.mana = 100
            player.hp = 100
        #Changing dictionary number to zero to prevent this event from activating again
        castle_info[player.movement_coordinate][1] = 0
    time.sleep(1)
    #Adding back door
    castle[player.movement_coordinate] = u'\U0001f6aa'
    return castle, player, castle_info

if __name__ == '__main__':
    print("You must run 'python3 start_game.py' to play Castle Descent.")   